### PR TITLE
feat: add component sub generator

### PR DESCRIPTION
**Pull request by jmfrancois**
To use it you just have to

```shell
$ yo react-cmf:component
```

```shell
     _-----_
    |       |    ╭──────────────────────────╮
    |--(o)--|    │ Welcome to the react-cmf │
   `---------´   │        generator!        │
    ( _´U`_ )    ╰──────────────────────────╯
    /___A___\   /
     |  ~  |
   __'.___.'__
 ´   `  |° ´ Y `

? name bar
? type (Use arrow keys)
❯ es6.class
  es6.arrow
  stateless
  createClass
```
then it create for you the following :
```shell
   create src/app/components/bar/bar.component.js
   create src/app/components/bar/bar.scss
   create src/app/components/bar/bar.tests.js
   create src/app/components/bar/index.js
```

the component look like this:
```javascript
import React from 'react';

class bar extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (<div>{this.props.name}</div>);
  }
}

bar.propTypes = {
  name: React.PropTypes.string,
};

export default bar;
```

### CODE REVIEW

File: generators/component/index.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+module.exports = yeoman.Base.extend({
+  prompting: function () {
+    // Have Yeoman greet the user.
+    this.log(yosay(
+      'Welcome to the react-cmf generator!'
+    ));
+
+    var prompts = [{
+      type: 'input',
+      name: 'name',
```
Comment (frassinier) [inline generators/component/index.js]: Do you plan to deal with spaces, numbers or special chars on name?

File: generators/component/templates/src/createClass.component.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+const <%= props.name %> = React.createClass({
+  propTypes: {
+    name: React.PropTypes.string,
+  },
+  render() {
+    return (<div>{this.props.name}</div>);
+  }
+});
+
+
```
Comment (frassinier) [inline generators/component/templates/src/createClass.component.js]: I think this new line is unnecessary

File: generators/component/templates/src/tests.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import <%= props.name %> from './<%= props.name %>.component';
+
+describe('<%= props.name %>', () => {
+  it('should render its name', () => {
+    const wrapper = shallow(<<%= props.name %> name="Hello world" />);
+    console.log(wrapper.debug());
```
Comment (frassinier) [inline generators/component/templates/src/tests.js]: Do you think this `console.log` is a good way to start? 
Maybe just the expect should be great?

File: generators/component/templates/src/scss [hunk 4]
```diff
@@ -0,0 +1 @@
+Component.scss
```
Comment (frassinier) [inline generators/component/templates/src/scss]: Where is the content?

File: generators/component/index.js (hunk 1 above)
Comment (jmfrancois) [inline generators/component/index.js]: That is a good question. Should a component being able to be named : "Mon super arré trié" ?
I don't think so.
It's a component name, so I don't think we should accept any space or special characters. But we should enforce a valid name. I will add that now.

**Pull request merged**